v2/pkg/ipam/pool.go

```diff
@@ -273,3 +284,15 @@ func (p *pool) AllocateBlock(ctx context.Context, nodeName string) (*coilv2.Addr
 	p.log.Error(ErrNoBlock, "no available blocks")
 	return nil, ErrNoBlock
 }
+
+// IsUsed returns true if the pool is used by some AddressBlock.
+func (p *pool) IsUsed(ctx context.Context) (bool, error) {
+	blocks := &coilv2.AddressBlockList{}
+	err := p.reader.List(ctx, blocks, client.MatchingLabels{
```

Comment:
Let me make sure. `isUsed` uses the direct API client, because `AddressPool` can be removed in the situation that a client returns no `AddressBlock`, but informer cache is staled(some AddressPools exists in fact). Is my understanding correct?

Reply:
@ysksuzuki 
The primary reason of using `p.reader` is that it is used in `pool.SyncBlocks()`.  I don't want to increase cached resources.

Typically the finalizer of `AddressPool` would be invoked in response to the deletion of the final `AddressBlock`.  `IsUsed()` should certainly return false, or else the finalizer will be delayed until the next forced reconcilliation.  So using cache in `IsUsed()` would be inappropriate.